src/pool/srv_pool.c

```diff
@@ -8234,6 +8234,74 @@ ds_pool_svc_stop_handler(crt_rpc_t *rpc)
 	pool_svc_stop_handler(rpc, DAOS_POOL_VERSION);
 }
 
+/* TODO: make a real RPC handler ds_pool_rebuild_stop_handler(crt_rpc *rpc) to call this. */
+void
+ds_pool_rebuild_stop(uuid_t pool_uuid, struct rsvc_hint *hint)
+{
+	struct pool_svc *svc;
+	int              rc;
+
+	D_INFO(DF_UUID ": received request to stop rebuild\n", DP_UUID(pool_uuid));
+
+	rc = pool_svc_lookup_leader(pool_uuid, &svc, hint);
+	if (rc != 0)
+		D_GOTO(out, rc);
+
+	ds_rebuild_admin_stop(pool_uuid);
+
+	/* TODO: ds_rsvc_set_hint(&svc->ps_rsvc, &out->hint); */
+	pool_svc_put_leader(svc);
+out:
+	/* TODO: RPC reply rc assign to rc */
+	D_INFO(DF_UUID ": rebuild stop result " DF_RC "\n", DP_UUID(pool_uuid), DP_RC(rc));
+}
+
+/* administrator (via dmg command) asks to resume normal rebuild behavior for pool
+ * (if it has not already happened due to another automatic or manual rebuild in the meantime).
+ * TODO: make a real RPC handler ds_pool_rebuild_start_handler(crt_rpc *rpc) to call this.
+ */
+void
+ds_pool_rebuild_start(uuid_t pool_uuid, struct rsvc_hint *hint)
+{
+	struct pool_svc *svc;
+	struct ds_pool  *pool;
+	daos_prop_t      prop = {0};
+	int              rc;
+
+	D_INFO(DF_UUID ": received request to resume/start rebuild\n", DP_UUID(pool_uuid));
+
+	rc = pool_svc_lookup_leader(pool_uuid, &svc, hint);
+	if (rc != 0)
+		D_GOTO(out, rc);
+
+	rc = ds_pool_lookup(pool_uuid, &pool);
```

Comment:
Why not use `svc->ps_pool`?

Reply:
Right, will do. I think it may have been left from an earlier prototyping form of the code where I had it implemented entirely within the rebuild module (which used ds_pool_svc_lookup_leader in the preceding call).